QModManager/API/ModLoading/Internal/ModLoadingResults.cs

```diff
@@ -0,0 +1,11 @@
+﻿namespace QModManager.API.ModLoading.Internal
+{
+    internal enum ModLoadingResults
+    {
+        Success,
+        NoMethodToExecute,
+        Failure,
+        AlreadyLoaded,
+        CurrentGameNotSupported,        
```

Comment:
@AlexejheroYTB As either a replacement or supplement to the current status enumns in this branch,   I'm proposing this `ModStatus` set to be used across the entirety of the mod loading life cycle.  
Once all loading is complete, we would have a final collection of all mods and their final status to report on in a single block.
```CSharp
internal enum ModStatus
{
  Success = 0,
  CanceledByAuthor = 1,
  PatchMethodFailed = 2,
  TooManyPatchMethods = 1,
  MissingPatchMethod = 2,
  CircularLoadOrder = 3,
  CircularDependency = 4,
  MissingDependency = 5,
  CurrentGameNotSupported = 6,
  DuplicateIdDetected = 7,
  MissingCoreData = 8,
  MissingAssemblyFile = 9
}
```
The values define how early the error happens in the loading life-cycle.  
Higher values are more critical.  
If we have a `Dictionary<QMod, ModStatus> LoadingStatus` somewhere, then we can always update it with something like this
```CSharp
LoadingStatus[mod] = Math.Max(LoadingStatus[mod], newStatus);
```
Which will keep the relevant status on top no matter what else we do with the QMod instance along the way.